map-comparison/DaskBenchmark.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+import dask.array
+import dask
+import scipy.ndimage
+
+from Benchmark import Benchmark
+
+class DaskBenchmark(Benchmark):
+    # 256 MB
+    target_chunksize=2**28
+
+    def __init__(self, path, dtype, scan_size, detector_size, warmup_rounds, roi, mask):
+        super().__init__(path, dtype, scan_size, detector_size, warmup_rounds, roi, mask)
+
+        # This assumes that scan_size[0] is powers of two
+        def calculate_chunking(target, dtype, scan_size, detector_size):
+            y = scan_size[0]
+            total_size = np.prod(scan_size + detector_size) * dtype.itemsize
+            ideal_num = total_size / target
+            real_num = 1
+            while real_num < ideal_num:
+                real_num *= 2
+
+            assert y % real_num == 0
+
+            return (real_num, (y // real_num, scan_size[1]) + detector_size)
+
+        def mmap_partition(filename, dtype, chunking, chunk_number):
+            chunksize = np.prod(chunking)*dtype.itemsize
+            offset = chunk_number * chunksize
+            return np.memmap(filename, dtype=dtype, mode='r', shape=chunking, offset=offset)
+
+        num, chunking = calculate_chunking(
+            self.target_chunksize, self.dtype, self.scan_size, self.detector_size
+        )
+        chunks = [dask.array.from_delayed(
+            dask.delayed(
+                mmap_partition(
```

Comment:
You are not really delaying the function call here; you could use `functools.partial` to supply the arguments, or return a function. I think something like this should work (untested):

```
def make_mmap_partition(filename, dtype, chunking, chunk_number):
    def _mmap_partition():
        chunksize = np.prod(chunking)*dtype.itemsize
        offset = chunk_number * chunksize
        return np.memmap(filename, dtype=dtype, mode='r', shape=chunking, offset=offset)
    return _mmap_partition
```

Reply:
Good point, thanks! Now I got it, I think.